querybuilder/rule.go

```diff
@@ -19,38 +19,45 @@ type Rule struct {
 }
 
 // Evaluate function checks whether the dataset matches with rule
-func (r *Rule) Evaluate(dataset map[string]interface{}) bool {
+func (r *Rule) Evaluate(dataset map[string]interface{}) (bool, error) {
+	var err error = nil
```

Comment:
i think this variable is not needed

Reply:
That variable stores the error returned by getInputValue() function.

Reply:
it can be declared when that method is called.